optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/bi/BiConstraintStream.java

```diff
@@ -771,6 +771,28 @@
      */
     <ResultA_> UniConstraintStream<ResultA_> map(BiFunction<A, B, ResultA_> mapping);
 
+    /**
+     * Takes each tuple and applies a mapping on the last fact, which turns it into {@link Iterable}.
+     * Returns a constraint stream consisting of tuples of the first fact
+     * and the contents of the {@link Iterable} one after another.
+     * This may produce a stream with duplicate tuples.
+     * See {@link #distinct()} for details.
+     *
+     * <p>
+     * In cases where the last fact is already {@link Iterable}, use {@link Function#identity()} as the argument.
+     *
+     * <p>
+     * Simple example: assuming a constraint stream of {@code (PersonName, Person)}
+     * {@code [Ann(roles = [USER, ADMIN]]), Beth(roles = [USER]), Cathy(roles = [ADMIN, AUDITOR])]},
```

Comment:
From this example, I can't see it's a `BiStream<String, Person>`. Maybe this would help illustrate its content:
```suggestion
     * {@code [(Ann, (name = Ann, roles = [USER, ADMIN])), (Beth, (name = Beth, roles = [USER])), (Cathy, (name = Cathy, roles = [ADMIN, AUDITOR]))]},
```